packages/core/src/waitFor.ts

```diff
@@ -44,10 +44,13 @@ export function waitFor<TActorRef extends ActorRef<any, any>>(
   };
   return new Promise((res, rej) => {
     let done = false;
-    const handle = setTimeout(() => {
-      sub.unsubscribe();
-      rej(new Error(`Timeout of ${resolvedOptions.timeout} ms exceeded`));
-    }, resolvedOptions.timeout);
+    const handle =
```

Comment:
```suggestion
    if (process.env.NODE_ENV !== 'production' && resolvedOptions.timeout < 0) {
        console.error('`timeout` passed to `waitFor` is negative and it will reject its internal promise immediately.')
    }
    const handle =
```